example/home/test/test_general.py

```diff
@@ -18,47 +35,62 @@ def query():
 
         blog_post = BlogPageFactory()
         another_post = BlogPageFactory()
-        results = self.client.execute(query())
+        results = self.client.execute(query, context_value=self.request)
 
         self.assertTrue("firstPost" in results["data"])
         self.assertEqual(int(results["data"]["firstPost"]["id"]), blog_post.id)
 
-        results = self.client.execute(
-            """
-            {
-                firstPost(order: "-id") {
-                    id
-                }
+        query = """
+        {
+            firstPost(order: "-id") {
+                id
             }
-            """
-        )
+        }
+        """
+        results = self.client.execute(query, context_value=self.request)
 
         self.assertTrue("firstPost" in results["data"])
         self.assertEqual(int(results["data"]["firstPost"]["id"]), another_post.id)
 
+    def test_singular_blog_page_query_with_user(self):
+        query = """
+        {
+            firstPost {
+                id
+            }
+        }
+        """
+        self.request.user = AuthenticatedUser()
+        results = self.client.execute(query, context_value=self.request)
+
+        data = results["data"]["firstPost"]
+        self.assertEqual(data, None)
```

Comment:
**Question :** do we really want the data to be `None` here ?

Reply:
@fabienheureux yes, because we have the [`IsAnonymousMiddleware`](https://github.com/GrappleGQL/wagtail-grapple/pull/177/files/cca22ad77fed9ea578e9e7d2567471b55b776f56#diff-d3f3dfeadc002df554ce4060d35a5ec429f66733c519d726ba80e447bc2e3062R72) being used on this query. So if we are authenticated, the query will return `None`.

Reply:
Ah. Makes sense. 
I assumed `IsAnonymousMiddleware` would also allow `Authenticated` users, but it is worthless as default queries are available to every users 😅